iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java

```diff
@@ -427,6 +427,10 @@ private TPipeConsensusTransferResp handleTransferFileSeal(final PipeConsensusTsF
       // loaded file. Since the writing file writer has already been closed, it will throw a Stream
       // Close exception.
       writingFileWriter.close();
+      // Sync here is necessary to ensure that the data is written to the disk. Or data region may
+      // load the file before the data is written to the disk and cause unexpected behavior after
+      // system restart. (e.g., empty file in data region's data directory)
+      writingFileWriter.getFD().sync();
```

Comment:
Will there be an exception when syncing after obtaining the FD after close?

Reply:
Not sure... Let me check the logs of CI ...